### ISSUE TITLE
Authentication error using scripted pipeline syntax

### ISSUE THREAD

**Issue by buep** [Praqma/pretested-integration-plugin#84]
I get an authentication error running with and adjusted example of documented example under `docs` folder.

Error is:

```
Started by an SCM change
[Pipeline] node
Running on master in /var/jenkins_home/workspace/pip-pipe
[Pipeline] {
[Pipeline] checkout
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/praqma-test/numbers.git # timeout=10
Fetching upstream changes from https://github.com/praqma-test/numbers.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials Credentail for ReleasePraqma
 > git fetch --tags --progress https://github.com/praqma-test/numbers.git +refs/heads/*:refs/remotes/origin/*
Seen branch in repository origin/master
Seen branch in repository origin/ready/buep-patch-4
Seen branch in repository origin/ready/buep-patch-5
Seen branch in repository origin/ready/buep-patch-7
Seen branch in repository origin/ready/buep-patch-8
Seen 5 remote branches
 > git tag -l # timeout=10
[PREINT]  Pretested Integration Plugin v3.0.0-SNAPSHOT (eaf4b)
[PREINT] Checking out integration branch master:
 > git config core.sparsecheckout # timeout=10
 > git checkout -f origin/master
 > git branch -a -v --no-abbrev # timeout=10
 > git branch -D master # timeout=10
 > git checkout -b master origin/master
[PREINT] Branch commit count: 1
[PREINT] Try FF as there is only one commit
 > git merge --ff-only d24857d20fae1ccfa95d86ae9d31c5b5677c4a94 # timeout=10
[PREINT] FF merge successful.
 > git rev-parse HEAD^{commit} # timeout=10
Seen branch in repository origin/master
Seen branch in repository origin/ready/buep-patch-4
Seen branch in repository origin/ready/buep-patch-5
Seen branch in repository origin/ready/buep-patch-7
Seen branch in repository origin/ready/buep-patch-8
Seen 5 remote branches
 > git tag -l # timeout=10
Checking out Revision d24857d20fae1ccfa95d86ae9d31c5b5677c4a94 (origin/ready/buep-patch-8)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f d24857d20fae1ccfa95d86ae9d31c5b5677c4a94
Commit message: "Update date-41.log"
First time build. Skipping changelog.
[Pipeline] sh
[pip-pipe] Running shell script
+ echo this is a build step
this is a build step
[Pipeline] pretestedIntegration
Triggered branch: origin/ready/buep-patch-8
integration branhc: master
repo: origin
uccredentials: 
[PREINT] Pushing changes to integration branch:
 > git config --get remote.origin.url # timeout=10
 > git --version # timeout=10
 > git push https://github.com/praqma-test/numbers.git HEAD:refs/heads/master
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
hudson.plugins.git.GitException: Command "git push https://github.com/praqma-test/numbers.git HEAD:refs/heads/master" returned status code 128:
stdout: 
stderr: remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/praqma-test/numbers.git/'

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1924)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1643)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:71)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$8.execute(CliGitAPIImpl.java:2003)
	at org.jenkinsci.plugins.gitclient.LegacyCompatibleGitAPIImpl.push(LegacyCompatibleGitAPIImpl.java:127)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.push(CliGitAPIImpl.java:71)
	at org.jenkinsci.plugins.gitclient.LegacyCompatibleGitAPIImpl.push(LegacyCompatibleGitAPIImpl.java:139)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.push(CliGitAPIImpl.java:71)
	at hudson.plugins.git.GitAPI.push(GitAPI.java:103)
	at org.jenkinsci.plugins.pretestedintegration.scm.git.GitBridge.pushToBranch(GitBridge.java:80)
	at org.jenkinsci.plugins.pretestedintegration.scm.git.GitBridge.pushToIntegrationBranchGit(GitBridge.java:68)
	at org.jenkinsci.plugins.pretestedintegration.PretestedIntegrationPostCheckout.perform(PretestedIntegrationPostCheckout.java:209)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:80)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1$1.call(SynchronousNonBlockingStepExecution.java:49)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1.run(SynchronousNonBlockingStepExecution.java:46)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE

Page generated: Oct 8, 2017 11:45:17 AM UTCREST APIJenkins ver. 2.73.1
```

The job configuration is as config.xml: 

```
<flow-definition plugin="workflow-job@2.14.1"><actions/><description/><keepDependencies>false</keepDependencies><properties><org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty><triggers><hudson.triggers.SCMTrigger><spec>* * * * *</spec><ignorePostCommitHooks>false</ignorePostCommitHooks></hudson.triggers.SCMTrigger></triggers></org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty></properties><definition class="org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition" plugin="workflow-cps@2.41"><script>

node {
   checkout([$class: 'GitSCM', branches: [[name: '*/ready/**']], doGenerateSubmoduleConfigurations: false, extensions: [gitPhlowIntegration(gitIntegrationStrategy: squash(), integrationBranch: 'master', repoName: 'origin')], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'ReleasePraqma', url: 'https://github.com/praqma-test/numbers.git']]])   
   sh 'echo "this is a build step"'
   pretestedIntegration()
}</script><sandbox>true</sandbox></definition><triggers/><disabled>false</disabled></flow-definition>
```

On the other hand the credentials work, because this job runs successfully and is a free-style job - config.xml here: 

```
<project><actions/><description/><keepDependencies>false</keepDependencies><properties/><scm class="hudson.plugins.git.GitSCM" plugin="git@3.6.0"><configVersion>2</configVersion><userRemoteConfigs><hudson.plugins.git.UserRemoteConfig><url>https://github.com/praqma-test/numbers.git</url><credentialsId>ReleasePraqma</credentialsId></hudson.plugins.git.UserRemoteConfig></userRemoteConfigs><branches><hudson.plugins.git.BranchSpec><name>*/ready/*</name></hudson.plugins.git.BranchSpec></branches><doGenerateSubmoduleConfigurations>false</doGenerateSubmoduleConfigurations><submoduleCfg class="list"/><extensions><org.jenkinsci.plugins.pretestedintegration.scm.git.PretestedIntegrationAsGitPluginExt plugin="pretested-integration@3.0.0-SNAPSHOT"><integrationBranch>master</integrationBranch><repoName>origin</repoName><gitIntegrationStrategy class="org.jenkinsci.plugins.pretestedintegration.scm.git.SquashCommitStrategy"/></org.jenkinsci.plugins.pretestedintegration.scm.git.PretestedIntegrationAsGitPluginExt></extensions></scm><canRoam>true</canRoam><disabled>true</disabled><blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding><blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding><triggers><hudson.triggers.SCMTrigger><spec>* * * * *</spec><ignorePostCommitHooks>false</ignorePostCommitHooks></hudson.triggers.SCMTrigger></triggers><concurrentBuild>false</concurrentBuild><builders><hudson.tasks.Shell><command>echo "build step - if successfull PIP will push to integration branch"</command></hudson.tasks.Shell></builders><publishers><org.jenkinsci.plugins.pretestedintegration.PretestedIntegrationPostCheckout plugin="pretested-integration@3.0.0-SNAPSHOT"/></publishers><buildWrappers/></project>
```

**Comment by buep** [comment]
I tested with latest master commit, local build: 

```
commit eaf4bab3ed106a1383c17cb1173d10cb80747310
Author: Bue Petersen <bue@praqma.net>
Date:   Fri Oct 6 13:11:20 2017 +0200

    For #83 updated pom to set ver. 3 and cleaned deps
    
    Bumped version to be 3.0.0 which will be next release.
    
    Also cleaned un-used plugin we don't use anymore in our build and delivery pipeline, so pom becomes simpler and mostly inherit quality checks from the parent pom.

```

**Comment by buep** [comment]
I get the same error with this setup, which is more correct:

```

stage ("Pretested Integration") {
node {
checkout([$class: 'GitSCM', branches: [[name: '*/ready/**']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'WipeWorkspace'], [$class: 'PruneStaleBranch'], gitPhlowIntegration(gitIntegrationStrategy: squash(), integrationBranch: 'master', repoName: 'origin')], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'ReleasePraqma-github-pass', url: 'https://github.com/praqma-test/numbers.git']]])
sh 'echo "here you should run build and simple test - your toll-gate criteria for integration"'
pretestedIntegration()
}
}
```

Error: 
```
Started by user admin
[Pipeline] stage
[Pipeline] { (Pretested Integration)
[Pipeline] node
Running on master in /var/jenkins_home/workspace/pip
[Pipeline] {
[Pipeline] checkout
Wiping out workspace first.
Cloning the remote Git repository
Cloning repository https://github.com/praqma-test/numbers.git
 > git init /var/jenkins_home/workspace/pip # timeout=10
Fetching upstream changes from https://github.com/praqma-test/numbers.git
 > git --version # timeout=10
using GIT_ASKPASS to set credentials ReleasePraqma-github-pass
 > git fetch --tags --progress https://github.com/praqma-test/numbers.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url https://github.com/praqma-test/numbers.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/praqma-test/numbers.git # timeout=10
Pruning obsolete local branches
Fetching upstream changes from https://github.com/praqma-test/numbers.git
using GIT_ASKPASS to set credentials ReleasePraqma-github-pass
 > git fetch --tags --progress https://github.com/praqma-test/numbers.git +refs/heads/*:refs/remotes/origin/* --prune
Seen branch in repository origin/master
Seen branch in repository origin/ready/buep-patch-8
Seen branch in repository origin/ready/buep-patch-9
Seen 3 remote branches
 > git tag -l # timeout=10
[PREINT]  Pretested Integration Plugin v3.0.0-SNAPSHOT (eaf4b)
[PREINT] Checking out integration branch master:
 > git config core.sparsecheckout # timeout=10
 > git checkout -f origin/master
 > git branch -a -v --no-abbrev # timeout=10
 > git checkout -b master origin/master
[PREINT] Branch commit count: 1
[PREINT] Try FF as there is only one commit
 > git merge --ff-only 4f83393355ba3d44ddad16b78792ce12988b07f0 # timeout=10
[PREINT] FF merge successful.
 > git rev-parse HEAD^{commit} # timeout=10
Seen branch in repository origin/master
Seen branch in repository origin/ready/buep-patch-8
Seen branch in repository origin/ready/buep-patch-9
Seen 3 remote branches
 > git tag -l # timeout=10
Checking out Revision 4f83393355ba3d44ddad16b78792ce12988b07f0 (origin/ready/buep-patch-9)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 4f83393355ba3d44ddad16b78792ce12988b07f0
Commit message: "Update numbers.txt"
 > git rev-list 4f83393355ba3d44ddad16b78792ce12988b07f0 # timeout=10
[Pipeline] sh
[pip] Running shell script
+ echo here you should run build and simple test - your toll-gate criteria for integration
here you should run build and simple test - your toll-gate criteria for integration
[Pipeline] pretestedIntegration
Triggered branch: origin/ready/buep-patch-9
integration branhc: master
repo: origin
uccredentials: 
[PREINT] Pushing changes to integration branch:
 > git config --get remote.origin.url # timeout=10
 > git --version # timeout=10
 > git push https://github.com/praqma-test/numbers.git HEAD:refs/heads/master
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
hudson.plugins.git.GitException: Command "git push https://github.com/praqma-test/numbers.git HEAD:refs/heads/master" returned status code 128:
stdout: 
stderr: remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/praqma-test/numbers.git/'

	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1924)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1643)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:71)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$8.execute(CliGitAPIImpl.java:2003)
	at org.jenkinsci.plugins.gitclient.LegacyCompatibleGitAPIImpl.push(LegacyCompatibleGitAPIImpl.java:127)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.push(CliGitAPIImpl.java:71)
	at org.jenkinsci.plugins.gitclient.LegacyCompatibleGitAPIImpl.push(LegacyCompatibleGitAPIImpl.java:139)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.push(CliGitAPIImpl.java:71)
	at hudson.plugins.git.GitAPI.push(GitAPI.java:103)
	at org.jenkinsci.plugins.pretestedintegration.scm.git.GitBridge.pushToBranch(GitBridge.java:80)
	at org.jenkinsci.plugins.pretestedintegration.scm.git.GitBridge.pushToIntegrationBranchGit(GitBridge.java:68)
	at org.jenkinsci.plugins.pretestedintegration.PretestedIntegrationPostCheckout.perform(PretestedIntegrationPostCheckout.java:209)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:80)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1$1.call(SynchronousNonBlockingStepExecution.java:49)
	at hudson.security.ACL.impersonate(ACL.java:260)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1.run(SynchronousNonBlockingStepExecution.java:46)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE
```



I tested it also on a 1.642.4 core (the one used in the pom.xml of the job) - there the pipeline job works fine.

I used this pipeline script on that version (Will not work on never, but it was generated the same way using the pipeline syntax generator).

```
checkout([$class: 'GitSCM', branches: [[name: '*/ready/**']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'WipeWorkspace'], [$class: 'PruneStaleBranch'], gitPhlowIntegration(gitIntegrationStrategy: squash(), integrationBranch: 'master', repoName: 'origin')], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'ReleasePraqma', url: 'https://github.com/praqma-test/numbers.git']]])
```

**Comment by MadsNielsen** [comment]
The issue is this: https://github.com/Praqma/pretested-integration-plugin/blob/eaf4bab3ed106a1383c17cb1173d10cb80747310/src/main/java/org/jenkinsci/plugins/pretestedintegration/PretestedIntegrationPostCheckout.java#L174

If you look in the console log, apparently with Workflow the credentials id is empty. If you look at the console log: 

`uccredentials:  ` <-- This indicates that the action does NOT have the information we need to authenticate for pipeline.

**Comment by buep** [comment]
@kryptag here was my steps in reproducing - sorry for not reporting them earlier:

`docker run -p 8080:8080 -p 50000:50000 jenkins/jenkins:2.73.1`

Then get the jenkins master up and running, remember to use the password echoed out to your console.
Then `mvn clean package` and go to Manage Jenkins -> Manage plugins and on last tab upload the `.hpi` file just build.

Then create the job, and before running configure credentials for a user and use that in the pipeline script.


I also tried with `docker run -p 8080:8080 -p 50000:50000 jenkins/jenkins:1.642.4` which neither doesn't work with credentials.

I both cases I also did a simple test with a freestyle build and the git scm extension, using the same credentials and there is seemed to work and integrated my ready branches.
